java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -139,9 +141,37 @@ private static Tree skipParentheses(Tree parent) {
     return skip;
   }
 
-  private static boolean isSubtype(Type expression, Type target) {
-    return expression.isSubtypeOf(target);
+  private static boolean isUnnecessarySubtypeCast(Type expressionType, TypeCastTree typeCastTree, Type target) {
+    return expressionType.isSubtypeOf(target) && !isLambdaCastNeeded(expressionType, typeCastTree, target);
   }
+
+  private static boolean isLambdaCastNeeded(Type expressionType, TypeCastTree typeCastTree, Type target) {
+    ExpressionTree expression = ExpressionUtils.skipParentheses(typeCastTree.expression());
+    if (expression.is(Tree.Kind.LAMBDA_EXPRESSION)) {
+      List<MethodJavaSymbol> childMethods = getMethodSymbolsOf(expressionType);
+      List<MethodJavaSymbol> parentMethods = getMethodSymbolsOf(target);
+
+      Predicate<JavaSymbol> isDefaultMethod = JavaSymbol::isDefault;
```

Comment:
Maybe make this as a field (even a static final one) ?

Reply:
yep, done